onadata/apps/logger/migrations/0028_add_user_to_daily_submission_counters.py

```diff
@@ -0,0 +1,68 @@
+from django.conf import settings
+from django.db import migrations, models
+from django.db.models import Subquery, deletion
+
+
+def add_user_to_daily_submission_counter(apps, schema_editor):
+    DailyXFormSubmissionCounter = apps.get_model("logger", "DailyXFormSubmissionCounter")
+    # delete any counters where xform and user are None, since we can't associate them with a user
+    DailyXFormSubmissionCounter.objects.filter(xform=None, user=None).delete()
+    # add the user to the counter, based on the xform user
+    DailyXFormSubmissionCounter.objects.all().exclude(xform=None).update(
+        user=Subquery(
+            DailyXFormSubmissionCounter.objects.all().exclude(xform=None).values('xform__user')[:1]
```

Comment:
I think this subquery is wrong. It updates all the row with the same user . 

```sql
UPDATE "logger_dailyxformsubmissioncounter"
   SET "user_id" = (
        SELECT U1."user_id"
          FROM "logger_dailyxformsubmissioncounter" U0
         INNER JOIN "logger_xform" U1
            ON (U0."xform_id" = U1."id")
         WHERE NOT (U0."xform_id" IS NULL)
         LIMIT 1
       )
 WHERE NOT ("logger_dailyxformsubmissioncounter"."xform_id" IS NULL)
 ```
 
Correct me if I'm wrong, but the subquery always returns the xform owner of the first daily counter.

I think you need to use a `OuterRef` over here. 

```python
DailyXFormSubmissionCounter.objects.all().exclude(xform=None).update(
    user=Subquery(
        DailyXFormSubmissionCounter.objects.filter(pk=OuterRef('pk')).values('xform__user')[:1]
    ),
)
```